src/http/routes.cc

```diff
@@ -168,13 +168,29 @@ routes& routes::put(operation_type type, const sstring& url, handler_base* handl
     if (it.second == false) {
         throw std::runtime_error(format("Handler for {} already exists.", url));
     }
+    _map_batch.emplace_back(std::make_pair(type,it.first));
     return *this;
 }
 
 match_rule* routes::del_cookie(rule_cookie cookie, operation_type type) {
     return delete_rule_from(type, cookie, _rules);
 }
 
+void routes::clean_batch() {
+    if (!_map_batch.empty()) {
+        for (auto& it : _map_batch) {
+            _map[it.first].erase(it.second);
```

Comment:
looks like you are removing all elements from `_map`. if that's the case, why not just use `_map.clear()`?

Reply:
This pr is not complete now. I just want to provide a batch for rules and remove the rules from the batch.

```cpp
// routes has 3
routes.open_batch()
routes.put(1);
routes2.put(2);
// routes has 1,2,3
routes.erase_batch();
// routes has 3
```